db/Pos/DB/Class.hs

```diff
@@ -183,13 +187,14 @@ gsMaxProposalSize = bvdMaxProposalSize <$> gsAdoptedBVData
 gsUnlockStakeEpoch :: MonadGState m => m EpochIndex
 gsUnlockStakeEpoch = bvdUnlockStakeEpoch <$> gsAdoptedBVData
 
+-- | Checks if provided epoch is in the bootstrap era (pure version)
+isBootstrapEraPure :: EpochIndex -> EpochIndex -> Bool
+isBootstrapEraPure unlockEpoch curEpoch =
+    not isDevelopment && isBootstrapEra unlockEpoch curEpoch
+
 -- | Checks if provided epoch is in the bootstrap era.
 gsIsBootstrapEra :: MonadGState m => EpochIndex -> m Bool
-gsIsBootstrapEra epoch =
-    if isDevelopment then pure False
-    else do
-        unlockStakeEpoch <- gsUnlockStakeEpoch
-        return $ isBootstrapEra unlockStakeEpoch epoch
+gsIsBootstrapEra epoch = flip isBootstrapEraPure epoch <$> gsUnlockStakeEpoch
```

Comment:
I like `do` version slightly more, because it makes it harder to confuse the order of arguments. But it's minor.